Source/Fuse.Reactive.Bindings/ItemsWindowList.uno

```diff
@@ -0,0 +1,328 @@
+					return GetDataKey(data, IdentityKey);
+					
+				case InstanceIdentity.Object:
+					return data;
+			}
+			
+			return null;
+		}
+
+		void WindowListItem.IDataWatcher.OnCurrentDataChanged( WindowListItem item, object oldData )
+		{
+			if (!_isRooted)
+				return;
+			var wi = item as T;
+			if (wi == null)
+			{
+				Fuse.Diagnostics.InternalError( "Invalid item in WindowList", this );
+				return;
+			}
+			
+			_listener.OnCurrentDataChanged( wi, oldData );
+		}
+		
+		protected override T CreateWindowItem( int dataIndex )
+		{
+			var data = GetData(dataIndex);
+			var wi = WindowListItem.Create<T>( this, GetDataId(data), data );
+			return wi;
+		}
+
+		protected override void OnErrorMessageChanged(string _errorMessage)
+		{
+			if (!_isRooted)
+				return;
+			if (_errorMessage != null)
+				_listener.SetFailed(_errorMessage);
+			else
+				_listener.SetValid();
+		}
+		
+		protected override void OnRemovedWindowItem(T wi) 
+		{ 
+			if (_isRooted) _listener.RemovedWindowItem(wi);
+		}
+		
+		protected override void OnAddedWindowItem(int windowIndex, T wi)
+		{
+			if (_isRooted) _listener.AddedWindowItem(windowIndex, wi);
+		}
+		
+		public override int GetDataCount()
+		{
+			//optimization to avoid needless logic in base class prior to rooting
+			if (!_isRooted) return 0;
+			
+			var e = _items as object[];
+			if (e != null) return e.Length;
+
+			var a = _items as IArray;
+			if (a != null) return a.Length;
+
+			return 0;
+		}
+		
+		
+		bool TryUpdateAt(int dataIndex, object newData)
+		{
+			if (Identity == InstanceIdentity.None)
+				return false;
+			
+			var windowIndex = DataToWindowIndex(dataIndex);
+			if (windowIndex < 0 || windowIndex >= WindowItemCount)
+				return false;
+				
+			var wi = GetWindowItem(windowIndex);
+			var newId = GetDataId(newData);
+			if (wi.Id == null || !Object.Equals(wi.Id, newId))
+				return false;
+				
+			wi.Data = newData;
+			return true;
+		}
+		
+		void PatchTo(IArray values)
+		{
+			//collect new ids in the window
+			var newIds = new List<object>();
+			var limit = CalcOffsetLimitCountOf(values.Length);
+			for (int i=0; i < limit; ++i)
+				newIds.Add( GetDataId(values[i+Offset]) );
+				
+			var curIds = new List<object>();
+			for (int i=0; i < WindowItemCount; ++i)
+				curIds.Add( GetWindowItem(i).Id);
+			
+			var ops = PatchList.Patch( curIds, newIds, PatchAlgorithm.Simple, null );
+			for (int i=0; i < ops.Count; ++i)
+			{
+				var op = ops[i];
+				switch (op.Op)
+				{
+					case PatchOp.Remove:
+						RemovedDataAt(op.A + Offset);
+						break;
+					case PatchOp.Insert:
+						InsertWindowItem(DataToWindowIndex(op.A + Offset), op.Data);
+						break;
+					case PatchOp.Update:
+						if (!TryUpdateAt(op.A + Offset, values[op.Data]))
+						{
+							RemovedDataAt(op.A + Offset);
+							InsertWindowItem(DataToWindowIndex(op.A + Offset), op.Data);
+						}
+						break;
+				}
+			}
+		}
+		
+		
+		void IObserver.OnSet(object newValue)
```

Comment:
`newValue` should not be used here?

Reply:
These interface values aren't necesarily needed. The underlying Observable has already gogten the new value, and `ItemWindowsList`  updates itself from the backing observable.

Should this be indicated somehow in this code? I agree it looks a bit weird.